parsing/location.mli

```diff
@@ -26,7 +26,27 @@ type t = Warnings.loc = {
   loc_start: Lexing.position;
   loc_end: Lexing.position;
   loc_ghost: bool;
-}
+  }
+(** [t] represents a segment of source code between two positions.
+
+    For positions coming from a parser, loc_ghost is usually false, which
+    indicates that the information attached to the location could in principle
+    be parsed from the location. This is used by tools like merlin that want to
+    relate source code with parsetrees or later asts.
```

Comment:
The "usually false" in the first proposition is non-informative , and you should make it clearer than you are documenting the `loc_ghost` field.

Maybe
```ocaml
 loc_ghost: bool;
  (**  [loc_ghost] is [false] if the location maps to a well-identified
        grammar construct in the source file. 
        If [loc_ghost] is [true], the location is a best-effort approximation for the
        origin of an elaborated construct that does not appear fully in the source.
*) }
(** [t] represents a segment of source code between two positions. *)
```
?